File: C/Easy/2000.c
char *reversePrefix(char *word, char ch)
{
    int length = strlen(word);
    char *answer = malloc((length + 1) * sizeof(char));

    int cond = 0;

    for (int i = 0; i < length; i++)
    {
        if (word[i] == ch)
        {
            cond = i;
            break;
        }
    }

    if (cond)
    {
        for (int j = 0; j <= cond; j++)
        {
            answer[j] = word[cond - j];
        }
        for (int k = cond + 1; k < length; k++)
        {
            answer[k] = word[k];
        }
    }
    else
    {
        return word;
    }
    answer[length] = '\0';

    return answer;
}

File: Java/Easy/2000.java
class Solution {
    public String reversePrefix(String word, char ch) {
        if (!word.contains(Character.toString(ch))) {
            return word;
        }
        int right = word.indexOf(ch);
        String answer = "";
        for (int i = right; i >= 0; i--) {
            answer += word.charAt(i);
        }
        for (int i = right + 1; i < word.length(); i++) {
            answer += word.charAt(i);
        }

        return answer;
    }
}